pkg/oauth/apis/oauth/validation/validation_test.go

```diff
@@ -746,3 +746,38 @@ func TestValidateAuthorizeTokensUpdate(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateUserNameField(t *testing.T) {
+	fieldArg := field.NewPath("metadata", "name")
+	valid := "kube:admin"
+	errs := ValidateUserNameField(valid, fieldArg)
+	if len(errs) != 0 {
+		t.Errorf("expected success: %v", errs)
+	}
+
+	errorCases := map[string]struct {
+		name string
+		T    field.ErrorType
+		F    string
+	}{
+		"system:admin": {
+			T: field.ErrorTypeInvalid,
+			F: "metadata.name",
+		},
+	}
+	for k, v := range errorCases {
+		errs := ValidateUserNameField(k, fieldArg)
+		if len(errs) == 0 {
+			t.Errorf("expected failure %s for %v", k, k)
```

Comment:
nit: more idiomatic would be to just say `t.Errorf("expected error, got none")`. If you `t.Run` the context will be clear in the output.

Reply:
if we literally expect just one error: `t.Error("expected 1 error, got %d", len(errs))`

Reply:
Done